src/ctelnet.cpp

```diff
@@ -544,82 +675,130 @@ void cTelnet::slot_socketDisconnected()
+                                 * modification for some locales, e.g. France, Spain.
+                                 */
+                                             .toString(tr("hh:mm:ss.zzz")));
     mNeedDecompression = false;
     reset();
 
-    if (!mpHost->isClosingDown()) {
-        postMessage(spacer);
-
+    if (!mpHost->isClosingDown() && mpSocket) {
+        // Don't do this if we are closing down - or we do not know which
+        // socket is the "active" one
 #if !defined(QT_NO_SSL)
-        QList<QSslError> sslErrors = getSslErrors();
-        QSslCertificate cert = mpSocket.peerCertificate();
+        if (mCurrent_sslTsl) {
+            // We were connecting/ed securely:
+            QList<QSslError> sslErrors = getSslErrors();
+            QSslCertificate cert = mpSocket->peerCertificate();
+            if (mpHost->mSslIgnoreExpired) {
+                sslErrors.removeAll(QSslError(QSslError::CertificateExpired, cert));
+            }
 
-        if (mpHost->mSslIgnoreExpired) {
-            sslErrors.removeAll(QSslError(QSslError::CertificateExpired, cert));
-        }
+            if (mpHost->mSslIgnoreSelfSigned) {
+                sslErrors.removeAll(QSslError(QSslError::SelfSignedCertificate, cert));
+            }
 
-        if (mpHost->mSslIgnoreSelfSigned) {
-            sslErrors.removeAll(QSslError(QSslError::SelfSignedCertificate, cert));
-        }
+            QStringList reasons;
+            if (mDontReconnect) {
+                // Do the same as for the insecure connection mode:
+                reasons.append(tr("User Disconnected"));
+            } else if (timeOffset >= 0 && timeOffset < 5000) {
+                reasons.append(tr("Connection/login attempt rejected by server"));
+            }
 
-        sslerr = (sslErrors.count() > 0 && !mpHost->mSslIgnoreAll && mpHost->mSslTsl);
+            sslerr = (sslErrors.count() > 0 && !mpHost->mSslIgnoreAll);
+            if (sslerr) {
+                // We also have SSL/TLS errors that are not ones we are ignoring
+                // so do not auto-reconnect:
+                mDontReconnect = true;
 
-        if (sslerr) {
-            mDontReconnect = true;
+                for (const auto& error : sslErrors) {
+                    reasons.append(error.errorString());
+                }
+            }
 
-            for (int a = 0; a < sslErrors.count(); ++a) {
-                reason.append(qsl("        %1\n").arg(QString(sslErrors.at(a).errorString())));
+            if (reasons.count()) {
+                postMessage(tr("[ ALERT ] - Socket got disconnected, for %n reason(s):\n"
+                               "%1",
+                               // Intentional comment to separate arguments
+                               "",
+                               reasons.count())
+                                    .arg(reasons.join(QChar::LineFeed)));
+            } else {
```

Comment:
<img width="583" height="44" alt="Image" src="https://github.com/user-attachments/assets/493c9772-ba39-4df5-b164-6cbd0eb0b543" />

This looks a bit awkward, what do you think about making a friendlier exception

```suggestion

            if (reasons.count() > 1) {
                postMessage(tr("[ ALERT ] - Socket got disconnected, for %n reason(s):\n"
                               "%1",
                               // Intentional comment to separate arguments
                               "",
                               reasons.count())
                                    .arg(reasons.join(QChar::LineFeed)));
            } else if (reasons.count() == 1) {
                postMessage(tr("[ ALERT ] - Socket got disconnected: "
                               "%1",
                               // Intentional comment to separate arguments
                               "")
                                    .arg(reasons.at(1));
            } else {
```

Reply:
:earth_africa: **I18n Not necessary!**

The `en` locales can be translated so that the `%n` value being `1` can indeed not mention the number of reasons. `linguist` will moan (a little) about not having an %n in the singular case but I'm not even sure that CrowdIn will object...

Other locales with different pluralisation rules are be different so extra C++ coding for a singular case is actually an anti-pattern IMVHO...

BTW the only reason for the `// Intentional comment...` and the `""` is because of the need to provide the third numeric arguement to the `QObject::tr(...)` method.

Reply:
The nittiest of nitpicks, but it can just be "reasons" now instead of "reason(s)"

Reply:
That is necessary in the "Engineering English" (source code) form but will be handled by the translation system when I ***don't*** make the change Vadi suggests.

Reply:
Aha, I get it now. My mistake!

Reply:
Agree with Vadi here. 
The english translation for (1) will still be like "Socket got disconnected, for a reason:\n" and this reads clunky.
How about "Socket got disconnected. Reason(s) is/are:\n" and don't even mention the number in the text.

Reply:
What, and avoid counting the number of fails? :raised_eyebrow::smile:

It is down to what fits for each locale - when there are multiple things it can be helpful to say how many there are - at least the `en_US` AND `en_GB` translations could just as easily be something of the form:
  `[ ALERT ] -  Socket got disconnected, because:\n`
  `             There is a problem between keyboard and chair.`
               
Or
  `[ ALERT ] -  Socket got disconnected, for 3 reasons:\n`
  `             The certificate was self-signed\n`
  `             The certificate expired on 2025/04/01\n`
  `             There is a problem between keyboard and chair.`

Reply:
Okay.